### DISCUSSION TITLE
hidden directory

### DISCUSSION THREAD

**Discussion by summermoon233** [HbmMods/Hbm-s-Nuclear-Tech-GIT#972] [Q&A]
Did the author really set the 4 puzzles for hidden directories  
Set the words of someone to solve it, in China, the answer with a translator translated out of the root can not see

**Comment by Antineutrino-4444** [comment]
where do you get the source?
as far as i know, no one have solved the catalog in china tho

**Reply by summermoon233** [reply]
MC encyclopedia, in China there are many people like to play HBM but this puzzle is not solved, and now we all think that the author did not set a puzzle
You are right, no one in China can solve this problem

**Reply by Antineutrino-4444** [reply]
u mean mcmod or what

**Reply by Billcyferka** [reply]
Brah literally check the source code the puzzle is there but the answers are so encrypted you cannot get them and yes there is an answer

**Reply by Antineutrino-4444** [reply]
dude we were talking about actually solving the puzzle, but decrypting the source code

**Reply by CirculatedBulletRain** [reply]
Decrypting is not possible at all unless you have the correct answer 
why is it impossible? it uses a hash encryption which is basically a one way encryption

**Reply by BallOfEnergy1** [reply]
The above is correct, decrypting isn't viable as it uses a SHA256 hashing algorithm.
I mean you can try, but you won't get far. 
(And yes, it'll be faster to solve it yourself instead of trying to brute force it even with a script)

**Reply by lilcocklil** [reply]
Maybe this directory us not even possible to get in survival mode. I think this riddles was made only for fun or just to make this mod more mystical and make him more famous. Why IT specialists dont look in the mods code, and if the allready did this why they didnt powdered answers in internet? Hidden directory is may be item what available only in creative mod like an button to call an meteorite or smthng like this idk.

**Reply by BallOfEnergy1** [reply]
It is possible. As an IT specialist who has looked in the mods code hundreds of times by this point, I can tell you why just looking in the code won't give the answers, and also that, if you do get the right answers, it *will* give you the item.

The code stores the "correct answers" as a set (multiple correct answers for variants of the answers) of salted SHA256 hashes. These hashes with this form of salting is extremely difficult to brute force. SHA256 avoids many collisions due to its bit size, so a hash collision is unlikely. Decrypting them doesn't make sense here, as a hash algorithm isn't really something that you *can* decrypt.

All of this is to say, looking in the code won't get you much.
